### PR TITLE
ABA numbers are now cross-checked with the file supplied

### PR DESCRIPTION
**Pull request by theblasher**
function was added in ApplicationUtility to read the file and tests were added

### CODE REVIEW

File: src/RoundTabler/NACHAScan.java [hunk 1]
```diff
@@ -1,5 +1,6 @@
 package RoundTabler;
 
+import java.util.ArrayList;
```
Comment (jordanmcm) [inline src/RoundTabler/NACHAScan.java]:
```suggestion
import java.util.HashSet;
```

File: src/RoundTabler/NACHAScan.java [hunk 2]
```diff
@@ -14,6 +15,7 @@
     private int pLastMatchStart;
     private int pLastMatchEnd;
     private String pLastMatchDescription;
+    private final ArrayList<String> abaNumbersList = getABANumbersFromFile();
```
Comment (jordanmcm) [inline src/RoundTabler/NACHAScan.java]:
```suggestion
    private final HashSet<String> abaNumbersList = getABANumbersFromFile();
```

File: src/RoundTabler/test/ApplicationUtilityTest.java [hunk 3]
```diff
@@ -42,4 +44,12 @@ public void charToIntPositiveTest(){
         int result3 = charToInt('3');
         assertEquals(3, result3);
     }
+
+
+    @Test
+    public void readFileTest() {
+        int expectedNumber = 19521;
+        ArrayList<String> abaNumbers = getABANumbersFromFile();
```
Comment (jordanmcm) [inline src/RoundTabler/test/ApplicationUtilityTest.java]:
```suggestion
        HashSet<String> abaNumbers = getABANumbersFromFile();
```

File: src/RoundTabler/test/ApplicationUtilityTest.java [hunk 4]
```diff
@@ -2,6 +2,8 @@
 
 import org.junit.jupiter.api.Test;
 
+import java.util.ArrayList;
```
Comment (jordanmcm) [inline src/RoundTabler/test/ApplicationUtilityTest.java]:
```suggestion
import java.util.HashSet;
```

File: src/utility/ApplicationUtility.java [hunk 5]
```diff
@@ -1,5 +1,10 @@
 package utility;
 
+import java.io.BufferedReader;
+import java.io.FileReader;
+import java.io.IOException;
+import java.util.ArrayList;
```
Comment (jordanmcm) [inline src/utility/ApplicationUtility.java]:
```suggestion
import java.util.HashSet;
```

File: src/utility/ApplicationUtility.java [hunk 6]
```diff
@@ -27,4 +32,28 @@ public static int charToInt(char character){
          */
         return (byte) character - 48;
     }
+
+    /**
+     * Forms an ArrayList of all ABA Numbers from the FedACHdir.txt file
+     * @return list of all abaNumbers from file
+     */
+    public static ArrayList<String> getABANumbersFromFile() {
+        ArrayList<String> abaNumbers = new ArrayList<>();
```
Comment (jordanmcm) [inline src/utility/ApplicationUtility.java]:
```suggestion
    public static HashSet<String> getABANumbersFromFile() {
        HashSet<String> abaNumbers = new HashSet<>();
```

**Review by jordanmcm — CHANGES_REQUESTED** [review]
A HashSet should be faster than an ArrayList implementation, but otherwise should operate the same in this case.
(Object.contains() is O(n) worse-case for ArrayList, O(1) for HashSet)

**Comment by theblasher** [comment]
changes were made

**Review by jordanmcm — APPROVED** [review]
I believe this works as intended and is an improvement over the previous implementation. For testing the 100% certainty, I inserted a new row into CustomerComments to test.
`INSERT INTO CustomerComments VALUES (9999999, '2022-11-13 16:57:28', 'randomuser', 'This should be shown as guaranteed under NACHA scan. 211371418');`
Which uses a number from the supplied ACH data.

**Pull request merged**